### ISSUE TITLE
opensnoop not displaying the full path to opened file

### ISSUE THREAD

**Issue by zhantt1** [iovisor/bcc#3124]
I built BCC from source on Ubuntu 18.04 and am running `opensnoop`. 

As a test, I open a file named `test.txt` with `vim` and wrote to it. The `opensnoop` output only displays the file name and not the full path to the opened file:

```
1001  23584  vim                 3   0 test.txt
``` 

I found another `opensnoop` output line that does display the full path to the opened file:

```
0     25828  splunkd            50   0 /opt/splunkforwarder/var/log/splunk/metrics.log
```

Any pointers on why only the file name was output in the first case? 

Also is there a method to filter the output such that only events from a certain file or directory are shown?

**Comment by yonghong-song** [comment]
> Any pointers on why only the file name was output in the first case?
You probably just did "vim test.txt"?

> Also is there a method to filter the output such that only events from a certain file or directory are shown?
You can do filtering in the bpf program (which is preferred) or do filtering in python program. bcc python tool has a helper `StrcmpRewrite` to facilitate to generate code for string compare in bpf programs, you may want to take a look.

**Comment by zhantt1** [comment]
Correct, that was the command I used. So to my understanding, if a relative path is used, then opensnoop will log the relative path? If I specify the full path with `vim`, opensnoop will report the full path to the file.  

Thank you for the pointer on filtering.

**Comment by yonghong-song** [comment]
opensnoop will only check argument and the full path is resolved inside the kernel. so yes, opensnoop will report whatever path in the syscall argument which is most likely provided by user.
